### PR TITLE
Config with c code

### PR DESCRIPTION
**Pull request by icemac**
- Add support for Python 3.9.
- Create wheels for aarch linux.
- Port the documentation to Python 3.

Fixes #24.

As `meta/config` has support for setting `PURE_PYTHON` it is done here, but this repository does not use it. Is this a problem?

### CODE REVIEW

File: docs/narr.rst [hunk 1]
```diff
@@ -98,17 +98,17 @@ object.  They cannot be changed once the message id is created:
...
   Traceback (most recent call last):
   ...
-  TypeError: readonly attribute
+  AttributeError: readonly attribute
 
   >>> robot.mapping = {u'name': u'Bender'}
   Traceback (most recent call last):
   ...
-  TypeError: readonly attribute
+  AttributeError: readonly attribute
```
Comment (icemac) [inline docs/narr.rst:111]: This is strange: We raise a TypeError in `__setattr__` but an `AttributeError` shows up here in Python 3.
Just PyPy3 still has the `TypeError` of Python 2. – That's really strange.

Raising an `AttributeError` in the codes does not break any other test. – Would this be an idea?

Comment (icemac) [reply docs/narr.rst:111]: `PyPy3` does not break on GHA because the Sphinx doctests are not run.

**Comment by icemac** [comment]
@mgedmin @jamadden Do you have an opinion on the change of the exception type or do you think this is just normal Python 3 behavior where PyPy3 is not (yet) compatible?

**Comment by jamadden** [comment]
This is a difference between the C and Python implementations. The Python version explicitly raises a `TypeError`, but because the C version doesn't specify any `tp_setattr` function, it gets the default, which raises `AttributeError`:
```console
$ python2.7 -c 'list().foo = 1'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'list' object has no attribute 'foo'
$ cat -p test.py
from zope.i18nmessageid import Message
Message('').foo = 1
$ python2.7 test.py
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    Message('').foo = 1
AttributeError: 'zope.i18nmessageid.message.Message' object has no attribute 'foo'
$ rm -f src/zope/i18nmessageid/_zope_i18nmessageid_message.so # This is the only way to go PURE_PYTHON
$ python2.7 test.py
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    Message('').foo = 1
  File "/zope.i18nmessageid/src/zope/i18nmessageid/message.py", line 80, in __setattr__
    raise TypeError('readonly attribute')
TypeError: readonly attribute
```
 I checked both Python 2.7 and Python 3.x, and they both behave this way. (So I don't know how the doctests ever passed; perhaps they haven't in a long time.) (Also, this package doesn't support the `PURE_PYTHON` environment variable 😞 )

I think the Python version should be changed to raise `AttributeError`. That will make it consistent with what most users have been seeing.

**Comment by icemac** [comment]
There is #32 which does the same and some additional portions if this PR where already ported to `master`, so closing this PR here.

**Pull request closed without merging**